actors/builtin/market/market_actor.go

```diff
@@ -633,10 +642,13 @@ func ValidateDealsForActivation(st *State, store adt.Store, dealIDs []abi.DealID
 	totalDealSpaceTime := big.Zero()
 	totalVerifiedSpaceTime := big.Zero()
 	for _, dealID := range dealIDs {
-		proposal, err := proposals.Get(dealID)
+		proposal, found, err := proposals.Get(dealID)
 		if err != nil {
 			return big.Int{}, big.Int{}, fmt.Errorf("failed to load deal %d: %w", dealID, err)
 		}
+		if !found {
+			return big.Int{}, big.Int{}, fmt.Errorf("dealId %d not found", dealID)
```

Comment:
Aside: I'm aware we'd like this to be ErrNotFound too, but can't propagate that information yet.